__tests__/pet.test.js

```diff
@@ -0,0 +1,292 @@
+const Pet = require('../src/pet');
+
+describe('constructor', () => {
+  let pet;
+  
+  beforeEach(() => {
+    pet = new Pet('Gerald');
+  });
+
+  it('returns an object', () => {
+    expect(new Pet('Gerald')).toBeInstanceOf(Object);
+  });
+
+  it('sets the name property', () => {
+    expect(pet.name).toEqual('Gerald');
+  });
+
+  it('has an initial age of 0', () => {
+    expect(pet.age).toEqual(0);
+  });
+
+  it('has an initial hunger of 0', () => {
+    expect(pet.hunger).toEqual(0);
+  });
+
+  it('has an initial fitness of 10', () => {
+    expect(pet.fitness).toEqual(10);
+  });
+});
+
+describe('growUp', () => {
+
+  beforeEach(() => {
+    pet = new Pet('Gerald');
+  });
+
+  it('increments the age by 1', () => {
+    pet.age = 0;
+    pet.growUp();
+
+    expect(pet.age).toEqual(1);
+  });
+
+  it('increments the hunger by 5', () => {
+    pet.hunger = 0;
+    pet.growUp();
+
+    expect(pet.hunger).toEqual(5);
+  });
+
+  it('decreases the fitness by 3', () => {
+    pet.fitness = 10;
+    pet.growUp();
+
+    expect(pet.fitness).toEqual(7);
+  });
+
+  it('throws an error message if the pet is no longer alive', () => {
+    pet.age = 30;
+    expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+  });
+});
+
+describe('walk', () => {
+
+  beforeEach(() => {
+    pet = new Pet('Gerald');
+  });
+
+  it('increments the fitness by 4, to a maximum of 10', () => {
+    pet.fitness = 8;
+    pet.walk();
+
+    expect(pet.fitness).toEqual(10);
+  });
+
+  it('throws an error message if the pet is no longer alive', () => {
+    pet.age = 30;
```

Comment:
In this case, I would set pet.fitness to 0, just cause you're testing the walk() function directly.